includes/admin/extensions/product-education/class-invoices.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Invoices
+ *
+ * Manages automatic installation/activation for Invoices.
+ *
+ * @package     EDD
+ * @subpackage  Invoices
+ * @copyright   Copyright (c) 2021, Easy Digital Downloads
+ * @license     https://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       2.11.x
+ */
+namespace EDD\Admin\Settings;
+
+use \EDD\Admin\Extensions\Extension;
+
+class Invoices extends Extension {
+
+	/**
+	 * The product ID on EDD.
+	 *
+	 * @var integer
+	 */
+	protected $item_id = 375153;
+
+	/**
+	 * The pass level required to access this extension.
+	 */
+	const PASS_LEVEL = \EDD\Admin\Pass_Manager::EXTENDED_PASS_ID;
+
+	public function __construct() {
+		$this->product_data = $this->get_product_data();
+		if ( ! $this->product_data ) {
+			return;
+		}
+		add_filter( 'edd_settings_sections_gateways', array( $this, 'add_section' ) );
+		add_action( 'edd_settings_tab_top_gateways_invoices', array( $this, 'settings_field' ) );
+		add_action( 'edd_settings_tab_top_gateways_invoices', array( $this, 'hide_submit_button' ) );
+
+		parent::__construct();
+	}
+
+	/**
+	 * Gets the custom configuration for Invoices.
+	 *
+	 * @since 2.11.x
+	 * @param array $product_data The array of product data from the API.
+	 * @return array
+	 */
+	protected function get_configuration( $product_data = array() ) {
+		return array(
+			'style'       => 'horizontal',
+			'title'       => __( 'Impress Your Customers with Custom Invoices', 'easy-digital-downloads' ),
+			'description' => __( 'Allow your customers to download beautiful, professional invoices with one click!', 'easy-digital-downloads' ),
+		);
+	}
+
+	/**
+	 * Adds the Invoices Payments section to the settings.
+	 *
+	 * @param array $sections
+	 * @return array
+	 */
+	public function add_section( $sections ) {
+		if ( $this->is_activated() ) {
```

Comment:
```suggestion
		if ( ! edd_is_admin_page( 'settings' ) ) {
			return $sections;
		}

		if ( $this->is_activated() ) {
```

If we add this to each extension class that checks `is_activated()` then we can prevent ever doing an API request unless the person is on the EDD settings page.